IaC/lambda_api_module/main.tf

```diff
@@ -0,0 +1,154 @@
+locals{
+  API_zipName = "../${api_name}/bin/Release/net6.0/${api_name}.zip"
+  APIFunctionName = "${api_name}Lambda"
+}
+
+resource "aws_iam_role" "LambdaRole" {
+  name = "${api_name}LambdaRole"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "LambdaPolicy" {
+  name        = "${api_name}LambdaPolicy"
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = [
+          "ec2:CreateNetworkInterface",
+          "ec2:DescribeNetworkInterfaces",
+          "ec2:DeleteNetworkInterface"
+        ]
+        Effect   = "Allow"
+        Resource = "*"
+      },
+      {
+          "Effect": "Allow",
+          "Action": [
+              "logs:CreateLogStream",
+              "logs:PutLogEvents"
+          ],
+          "Resource": [
+              "${aws_cloudwatch_log_group.LambdaLogGroup.arn}:*"
+          ]
+      }
+    ]
+  })
+}
+
+resource "aws_iam_policy_attachment" "LambdaPolicyAttachment" {
+  name       = "${api_name}LambdaPolicyAttachment"
+  roles      = [aws_iam_role.LambdaRole.name]
+  policy_arn = aws_iam_policy.LambdaPolicy.arn
+}
+
+
+
+resource "aws_lambda_function" "APILambda" {
+  depends_on = [
+    aws_iam_policy_attachment.LambdaPolicyAttachment
+  ]
+  filename      = local.API_zipName
+  function_name = local.APIFunctionName
+  role          = aws_iam_role.LambdaRole.arn
+  handler       = "bootstrap"
+  runtime = "provided.al2"
+  timeout = 60
+
+  source_code_hash = filebase64sha256(local.API_zipName)
+  vpc_config {
```

Comment:
These need to be variables because :

* this file is the source of the child module
* these resources are coming from the root module
* the child module knows nothing about the resources in the root module (or about the root module itself)